app/experimenter/nimbus-ui/src/index.test.tsx

```diff
@@ -102,3 +73,30 @@ describe("index", () => {
     expect(mockError).toHaveBeenCalled();
   });
 });
+
+// Mock out these components - for testing, we only care that they're used
+jest.mock("react", () => ({
+  ...jest.requireActual("react"),
+  StrictMode: mockComponent("StrictMode"),
+}));
+
+jest.mock("./components/AppErrorBoundary", () => ({
+  __esModule: true,
+  default: mockComponent("AppErrorBoundary"),
+}));
+
+jest.mock("@apollo/client", () => ({
+  ...jest.requireActual("@apollo/client"),
+  ApolloProvider: mockComponent("ApolloProvider"),
+}));
+
+jest.mock("./components/App", () => ({
+  __esModule: true,
+  default: mockComponent("App"),
+}));
+
+function mockComponent(testid: string) {
```

Comment:
Oh, and this is an annoying thing I discovered: I really wanted to DRY up the little functional components in those `jest.mock()` calls, maybe move that `mockComponent` function into a shared module.

As it turns out, though, Jest uses a babel plugin to reorder the code at compile time. All the `jest.mock` calls are moved to the top of the syntax tree before all import statements in order to ensure the mocks happen before loading the mocked modules. That means these mocks can't use imported code. So, things are only partly DRY-er